homeassistant/components/devolo_home_network/button.py

```
@@ -0,0 +1,132 @@
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all devices and buttons and setup them via config entry."""
+    device: Device = hass.data[DOMAIN][entry.entry_id]["device"]
+
+    entities: list[DevoloButtonEntity] = []
+    if device.plcnet:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[IDENTIFY],
+                device,
+            )
+        )
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[PAIRING],
+                device,
+            )
+        )
+    if device.device and "restart" in device.device.features:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[RESTART],
+                device,
+            )
+        )
+    if device.device and "wifi1" in device.device.features:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[START_WPS],
+                device,
+            )
+        )
+    async_add_entities(entities)
+
+
+class DevoloButtonEntity(DevoloEntity, ButtonEntity):
+    """Representation of a devolo button."""
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        description: DevoloButtonEntityDescription,
+        device: Device,
+    ) -> None:
+        """Initialize entity."""
+        self.entity_description: DevoloButtonEntityDescription = description
+        super().__init__(entry, device)
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        try:
+            await self.entity_description.press_func(self.device)
+        except DevicePasswordProtected:
+            self.entry.async_start_reauth(self.hass)
+        except DeviceUnavailable:
+            _LOGGER.error("Device %s did not respond", self.entry.title)
```

This should raise `HomeAssistantError` instead of logging so user get's feedback